Marketplace/voiceHUDApp/appResources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,836 @@
+                Overlays.editOverlay(id, { color: {
+                    "red": 0,
+                    "green": 255,
+                    "blue": 0
+                } });
+            });
+        } else if (user.audioLevel > 0.5 && user.audioLevel <= 0.75){ 
+            user.audioBarIDs.forEach(function(id){
+                Overlays.editOverlay(id, { color: {
+                    "red": 255,
+                    "green": 255,
+                    "blue": 0
+                } });
+            });
+        } else if (user.audioLevel > 0.75){ 
+            user.audioBarIDs.forEach(function(id){
+                Overlays.editOverlay(id, { color: {
+                    "red": 255,
+                    "green": 0,
+                    "blue": 0
+                } });
+            });
+        }
+        user.audioBarIDs.forEach(function(index){                
+            var dimensionsWithSound = HUD_MIN_Z_DIMENSIONS + user.audioLevel * 
+                (HUD_MAX_Z_DIMENSIONS[user.audioBarIDs.indexOf(index)] - HUD_MIN_Z_DIMENSIONS);
+            Overlays.editOverlay(index, { 
+                dimensions: { x: HUD_DEFAULT_DIMENSIONS.x, y: HUD_DEFAULT_DIMENSIONS.y, z: dimensionsWithSound } 
+            });
+        });            
+    }
+
+
+    // Set the bars of the HUD to the default size
+    function setDefaultHUD(uuid) {
+        var user = userStore[uuid];
+        user.audioBarIDs.forEach(function(id){
+            Overlays.editOverlay(id, { dimensions: HUD_DEFAULT_DIMENSIONS });
+        });   
+    }
+
+
+    // The function removes all HUDs from all users
+    function removeAll() {
+        // remove previous overlays
+        for (var i = 0; i < settings.users.length; i++) {
+            var user = settings.users[i];
+            var uuid = user.uuid;
+            deleteHUD(uuid);
+        }
+        var overlayList = Overlays.findOverlays(MyAvatar.position, 1000);
+        overlayList.forEach(function(overlay){
+            if (Overlays.getProperty(overlay, "name") === "Voice-HUD"){
+                Overlays.deleteOverlay(overlay);
+            }
+        });
+        overlayList = null;
+    }
+
+    // This function adds HUDs for all users in the PAL list
+    function addAll() {
+        // loook for old overlays to clean up:
+        var overlayList = Overlays.findOverlays(MyAvatar.position, 1000);
+        overlayList.forEach(function(overlay){
+            if (Overlays.getProperty(overlay, "name") === "Voice-HUD"){
+                Overlays.deleteOverlay(overlay);
+            }
+        });
+        overlayList = null;
+
+        if (!settings.users.length) {
+            // only add people to the list if there are none
+            sortData();
+        }
+
+        // add new overlays
+        for (var i = 0; i < settings.users.length; i++) {
+            var user = settings.users[i];
+            var uuid = user.uuid;
+            createHUD(uuid);
+        }
+    }
+
+
+    // This holds the PAL data
+    var settings = {
```

Comment:
why is this called `settings`? it doesn't seem to have anything to do with settings to me - the name is confusing to me. and what's the difference between `settings` and `userStore`?

Reply:
Remnants of Robin's code, changed to userArray and userObject respectively, until I determine that I can delete one altogether.